src/cursor_on_target.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""
+cursor_on_target.py
+
+This module implements Cursor on Target (CoT) functionality for the OpenAthena project.
+It provides functions to create, format, and send CoT messages over a UDP multicast network.
+
+CoT is a XML-based schema for sharing location information between different systems.
+This implementation focuses on sending location data from drone imagery analysis.
+
+The module uses the following libraries:
+- xml.etree.ElementTree for XML creation
+- uuid for generating unique identifiers
+- time for timestamping
+- socket for network communication
+
+Usage:
+    Import this module and use the create_and_send_cot function to send CoT messages
+    with resolved target information.
+
+Note:
+    This implementation uses UDP multicast by default, which may not be suitable
+    for all network environments. Ensure your network supports multicast before use.
+"""
+
+import xml.etree.ElementTree as ET
+import uuid
+import time
+import socket
+#import struct
+import socket
+import logging
+
+"""
+    Generate a unique identifier for CoT messages.
+
+    Returns:
+    str: A unique identifier string prefixed with 'OpenAthena-'.
+
+    Note:
+    This function uses UUID4 to ensure uniqueness across multiple instances
+    and executions of the software.
+"""
+def create_uid():
+    return f"OpenAthena-{uuid.uuid4()}"
+
+"""
+    Construct a CoT XML message.
+
+    Parameters:
+    lat (float): Latitude of the target in decimal degrees.
+    lon (float): Longitude of the target in decimal degrees.
+    alt (float): Altitude of the target in meters above WGS84 ellipsoid.
+    ce (float): Circular error of the position estimate in meters.
+    le (float): Linear error of the altitude estimate in meters.
+
+    Returns:
+    str: A formatted XML string representing the CoT message.
+
+    Note:
+    The function sets the CoT event type to 'a-p-G' (Assumed friend - Pending - Ground)
+    and the 'how' attribute to 'h-c' (Human - Calculated).
+"""
+def build_cot_xml(lat, lon, alt, ce, le):
+    root = ET.Element("event")
+    root.set("version", "2.0")
+    root.set("uid", create_uid())
+    root.set("type", "a-p-G")
+    root.set("how", "h-c")
+    root.set("time", time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()))
```

Comment:
Would recommend using image EXIF time for this field, e.g. here:

https://github.com/Theta-Limited/ThetaTools/blob/c62c02c33681b5ea072c106558f4c24f47963f8a/CursorOnTargetSender.java#L104